### ISSUE TITLE
Weird (Incorrect?) URLParam results

### ISSUE THREAD

**Issue by DylanJ** [go-chi/chi#100]
I am not sure if this is a bug, or me just not using the routing correctly. I have some odd routes:

``` go
    r.Route("/api/v1", func(r chi.Router) {
        r.Route("/files", func(r chi.Router) {
            r.Get("/:foo/hello/:bar", a)
            r.Get("/:bar/world/:baz", b)
            r.Get("/:bar/sups/:baz/:foo", c)
        })
    })
```

chi correctly routes the request, however the params are not set correctly. For example a request to `/api/v1/files/a/sups/b/c` correctly routes to `func c()` however the decoded params look like `{"bar":"","baz":"b","foo":"a","req":"c"}`

Here is an example with failing tests:

https://gist.github.com/DylanJ/4d3c7d1ea6f99255df823be717e99bcc

**Comment by pkieltyka** [comment]
Looks similar to https://github.com/pressly/chi/issues/78

**Comment by donutloop** [comment]
I have the same issue

**Comment by pkieltyka** [comment]
solved by v3 release 02e6bbdfd14640d8c1beb15ec3a38140400fb5c2 

See complete CHANGELOG for v3: https://github.com/go-chi/chi/blob/master/CHANGELOG.md
